### PR TITLE
Derive SequencerClient

### PR DESCRIPTION
**Pull request by jfldde**
# Description

- Use jsonrpsee proc macro to derive SequencerClient trait
- Refactor to use `rpc(client, server)`
- Explicilty mark method behind test_mode as 404

### CODE REVIEW

File: crates/sequencer/src/rpc.rs [hunk 1]
```diff
@@ -108,39 +131,64 @@ pub(crate) fn create_rpc_module<
...
-
-    rpc.register_blocking_method(
-        "citrea_sendRawDepositTransaction",
-        move |parameters, ctx, _| {
-            let mut params = parameters.sequence();
-            let deposit: Bytes = params.next()?;
+    }
 
-            debug!("Sequencer: citrea_sendRawDepositTransaction");
+    fn citrea_send_raw_deposit_transaction(&self, deposit: Bytes) -> RpcResult<()> {
```
Comment (rakanalh) [inline crates/sequencer/src/rpc.rs:136]:
```suggestion
    fn send_raw_deposit_transaction(&self, deposit: Bytes) -> RpcResult<()> {
```

File: crates/sequencer/src/rpc.rs [hunk 2]
```diff
@@ -108,39 +131,64 @@ pub(crate) fn create_rpc_module<
...
+                Ok(())
+            }
+            Err(e) => {
+                error!("Error processing deposit tx: {:?}", e);
+                Err(e.into())
             }
+        }
+    }
+
+    async fn citrea_test_publish_block(&self) -> RpcResult<()> {
```
Comment (rakanalh) [inline crates/sequencer/src/rpc.rs:166]:
```suggestion
    async fn publish_block(&self) -> RpcResult<()> {
```

**Review by rakanalh — APPROVED** [review]
NICE!

File: crates/sequencer/src/rpc.rs (hunk 2 above)
Comment (rakanalh) [reply crates/sequencer/src/rpc.rs:166]: Maybe `publish_test_block` would be okay

**Review by ercecan — APPROVED** [review]
🔥

**Pull request merged**